### PR TITLE
CsvReader from csvwriter and csvanalyzer

### PR DESCRIPTION
**Pull request by neubs-bsi**
#404 
This combines the csv analyzer functionality of reading a csv file and mapping it on an artifact with the SW360ExporterCSVWriter functionality to produce one. 

Several places needed adjustment to create a roundtrip. 

@blaumeiser-at-bosch @bs-ondem 

*Type of change*:  improvement

Tests were moved to new corresponding classes. 

### Checklist
Must:
- [x] All related issues are referenced in commit messages

Optional: *(delete if not applicable)*
- [x] I have provided tests for the changes (if there are changes that need additional tests)

### CODE REVIEW

File: modules/sw360/pom.xml [hunk 1]
```diff
@@ -159,6 +159,10 @@
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
         </dependency>
+        <dependency>
```
Comment (blaumeiser-at-bosch) [inline modules/sw360/pom.xml]: Why do you need the simple slf4j backend implementation? Is this a test dependency?

Comment (neubs-bsi) [reply modules/sw360/pom.xml]: This gets rid of the warning of the missing SLFJ4 warning when executing the tests. 
I can remove this though and do it in a separate pull request.

Comment (blaumeiser-at-bosch) [reply modules/sw360/pom.xml]: So you could add it as a testing dependency?

Comment (neubs-bsi) [reply modules/sw360/pom.xml]: I think this occurs when executing the classes in general, it just is visible when executing the tests.
I'd rather check this out later again and remove the change for now, if you don't mind.

**Comment by neubs-bsi** [comment]
TODO in code refers to #405

**Pull request merged**